src/main/java/org/broadinstitute/hellbender/tools/copynumber/segmentation/MultidimensionalKernelSegmenter.java

```diff
@@ -0,0 +1,188 @@
+
+        MultidimensionalPoint(final SimpleInterval interval,
+                              final double log2CopyRatio,
+                              final double alternateAlleleFraction) {
+            this.interval = interval;
+            this.log2CopyRatio = log2CopyRatio;
+            this.alternateAlleleFraction = alternateAlleleFraction;
+        }
+
+        @Override
+        public String getContig() {
+            return interval.getContig();
+        }
+
+        @Override
+        public int getStart() {
+            return interval.getStart();
+        }
+
+        @Override
+        public int getEnd() {
+            return interval.getEnd();
+        }
+    }
+
+    private final CopyRatioCollection denoisedCopyRatios;
+    private final OverlapDetector<CopyRatio> copyRatioMidpointOverlapDetector;
+    private final AllelicCountCollection allelicCounts;
+    private final OverlapDetector<AllelicCount> allelicCountOverlapDetector;
+    private final Map<String, List<MultidimensionalPoint>> multidimensionalPointsPerChromosome;
+
+    public MultidimensionalKernelSegmenter(final CopyRatioCollection denoisedCopyRatios,
+                                           final AllelicCountCollection allelicCounts) {
+        Utils.nonNull(denoisedCopyRatios);
+        Utils.nonNull(allelicCounts);
+        Utils.validateArg(denoisedCopyRatios.getSampleName().equals(allelicCounts.getSampleName()),
+                "Sample names do not match.");
+        this.denoisedCopyRatios = denoisedCopyRatios;
+        copyRatioMidpointOverlapDetector = denoisedCopyRatios.getMidpointOverlapDetector();
+        this.allelicCounts = allelicCounts;
+        allelicCountOverlapDetector = allelicCounts.getOverlapDetector();
+        final int numAllelicCountsToUse = (int) denoisedCopyRatios.getRecords().stream()
+                .filter(allelicCountOverlapDetector::overlapsAny)
+                .count();
+        logger.info(String.format("Using first allelic-count site in each copy-ratio interval (%d / %d) for multidimensional segmentation...",
+                numAllelicCountsToUse, allelicCounts.size()));
+        multidimensionalPointsPerChromosome = denoisedCopyRatios.getRecords().stream()
+                .map(cr -> new MultidimensionalPoint(
+                        cr.getInterval(),
+                        cr.getLog2CopyRatioValue(),
+                        allelicCountOverlapDetector.getOverlaps(cr).stream()
+                                .min(SampleLocatableCollection.LEXICOGRAPHICAL_ORDER_COMPARATOR::compare)
+                                .orElse(BALANCED_ALLELIC_COUNT).getAlternateAlleleFraction()))
+                .collect(Collectors.groupingBy(
+                        MultidimensionalPoint::getContig,
+                        LinkedHashMap::new,
+                        Collectors.toList()));
+    }
+
+    public MultidimensionalSegmentCollection findSegmentation(final int maxNumChangepointsPerChromosome,
```

Comment:
Missing javadoc

Reply:
Good catch, added some here and to the other KernelSegmenters.